source/includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go

```diff
@@ -0,0 +1,151 @@
+	clientOptions := options.Client().ApplyURI(uri)
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		log.Fatalf("failed to connect to the server: %v", err)
+	}
+	defer func() { _ = client.Disconnect(ctx) }()
+
+	// Set the namespace
+	coll := client.Database("sample_mflix").Collection("embedded_movies")
+
+	// start-create-vector-search
+	// Defines the structs used for the index definition
+	type vectorDefinitionField struct {
+		Type          string `bson:"type"`
+		Path          string `bson:"path"`
+		NumDimensions int    `bson:"numDimensions"`
+		Similarity    string `bson:"similarity"`
+		Quantization  string `bson:"quantization"`
+	}
+
+	type vectorDefinition struct {
+		Fields []vectorDefinitionField `bson:"fields"`
+	}
+
+	// Sets the index name and type to "vectorSearch"
+	indexName := "vector_index"
+	opts := options.SearchIndexes().SetName(indexName).SetType("vectorSearch")
+
+	// Defines the index definition
+	vectorSearchIndexModel := mongo.SearchIndexModel{
+		Definition: vectorDefinition{
+			Fields: []vectorDefinitionField{{
+				Type:          "vector",
+				Path:          "plot_embedding",
+				NumDimensions: 1536,
+				Similarity:    "dotProduct",
+				Quantization:  "scalar"}},
+		},
+		Options: opts,
+	}
+
+	// Creates the index
+	searchIndexName, err := coll.SearchIndexes().CreateOne(ctx, vectorSearchIndexModel)
+	if err != nil {
+		log.Fatalf("failed to create the vector search index: %v", err)
+	}
+	// end-create-vector-search
+
+	// Creates an Atlas Search index
+	// start-create-atlas-search
+	// Sets the index name and type to "search"
+	indexName := "atlas_search_index"
+	opts := options.SearchIndexes().SetName(indexName).SetType("search")
+
+	// Defines the index definition 
+	searchIndexModel := mongo.SearchIndexModel{
+		Definition: bson.D{
+			{Key: "mappings", Value: bson.D{
+				{Key: "dynamic", Value: false},
+				{Key: "fields", Value: bson.D{
+					{Key: "plot", Value: bson.D{
+						{Key: "type", Value: "string"},
+					}},
+				}},
+			}},
+		},
+		Options: opts,
+	}
+
+	// Creates the index
+	searchIndexName, err := coll.SearchIndexes().CreateOne(ctx, searchIndexModel)
+	if err != nil {
+		log.Fatalf("failed to create the atlas search index: %v", err)
+	}
+	// end-create-atlas-search
+
+	// start-list-index
+	// Specify the index to retrieve
+	indexName := "<indexName>"
+	opts := options.SearchIndexes().SetName(indexName)
+
+	// Retrieves the details of the specified index
+	cursor, err := coll.SearchIndexes().List(ctx, opts)
+
+	// Prints the index details to the console as JSON
+	var results []bson.M
+	if err := cursor.All(ctx, &results); err != nil {
+		log.Fatalf("failed to unmarshal results to bson: %v", err)
+	}
+	res, err := json.Marshal(results)
+	if err != nil {
+		log.Fatalf("failed to marshal results to json: %v", err)
+	}
+	fmt.Println(res)
+	// end-list-index
+
+	// start-update-index
+	// Specify the index name and the new index definition 
```

Comment:
```suggestion
	// Specifies the index name and the new index definition 
```
Nit: Keeps it in the same tense as the rest of the comments

Reply:
@shuangela According to this [guidance on code comments](https://docs.google.com/document/d/1Bnhh1OsGRUuXo_6yVUASGWiSX6sWuqrEFxPP6sLZDHM/edit?tab=t.0#heading=h.xv6glj2lnzjh), comments indicating for a user to take an action should be imperative. For this section, the user needs to replace the index name and definition, so the imperative tense is necessary here.

Reply:
Yes, the delete, list, and update methods are used the same for both AS and AVS. I updated the examples to remove the placeholders and include more general names, like myIndex. The only exception is in the update section where I made the name vector_index, since the definition shown is for a vector index.
Also changed the comments back to "specifies".

Reply:
Ok, that makes sense. Thanks for making the change.

Any reason you decided to not make the names `atlas_search_index` and `vector_index` more consistent though? I would either use "atlas" for both or neither. Also, the corresponding terms are "search" vs "vector search" rather than "search" vs "vector". Let me know if there's something I'm missing

Reply:
Thanks for clarifying! Updated names to ``vector_search_index`` and ``search_index``